crud/common/sharding.lua

```diff
@@ -20,7 +22,17 @@ function sharding.tuple_get_bucket_id(tuple, space, specified_bucket_id)
         return specified_bucket_id
     end
 
-    local key = utils.extract_key(tuple, space.index[0].parts)
+    local primary_index = space.index[0]
+    local key
+    local sharding_key = sharding.get_ddl_sharding_key(space.name)
+    if sharding_key ~= nil then
+        local space_format = space:format()
+        local sharding_key_fieldno_map = utils.get_keys_fieldno_map(space_format, sharding_key)
+        key = utils.extract_sharding_key(tuple, primary_index.parts, sharding_key_fieldno_map)
```

Comment:
How does it work? As far I understand this function doesn't consider an order of parts. So it looks like with primary key - {a, b, c, d, e} and sharing keys {a, b, c} and {c, b, a} work in the same way. But it's not so.

```
tarantool> vshard.router.bucket_id_mpcrc32({1, 2, 3})
---
- 20530
...

tarantool> vshard.router.bucket_id_mpcrc32({3, 2, 1})
---
- 14700
...
```

Why don't tests catch it?

Reply:
> How does it work? 

exactly as ` utils.extract_key()` but exclude fields not specified in sharding key.